api/grpcserver/transaction_service.go

```diff
@@ -63,16 +66,20 @@ func (s TransactionService) SubmitTransaction(ctx context.Context, in *pb.Submit
 	}
 
 	if !s.syncer.IsSynced(ctx) {
-		return nil, status.Error(codes.FailedPrecondition,
-			"Cannot submit transaction, node is not in sync yet, try again later")
+		return nil, status.Error(codes.FailedPrecondition, "Cannot submit transaction, node is not in sync yet, try again later")
+	}
+
+	if err := s.txHandler.VerifyAndCacheTx(ctx, in.Transaction); err != nil {
+		log.Error("error verifying incoming tx: %v", err)
+		return nil, status.Error(codes.InvalidArgument, "Failed to verify transaction")
```

Comment:
i think it is better to return the error and drop the log, as it will be printed grpc middleware anyway